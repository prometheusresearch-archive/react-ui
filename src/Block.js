/**
 * @copyright 2016-present, Prometheus Research, LLC
 * @flow
 */

import * as React from 'react';
import {chooseValue} from './Utils';
import * as theme from './theme';
import type {I18NContext} from './I18N';
import * as I18N from './I18N';

type Size = number | string;

type Props = {
  Component?: string | Function,
  inline?: boolean,
  noWrap?: boolean,
  position?: 'relative' | 'absolute' | 'fixed',
  width?: Size,
  maxWidth?: Size,
  minWidth?: Size,
  height?: Size,
  maxHeight?: Size,
  minHeight?: Size,
  top?: Size,
  left?: Size,
  bottom?: Size,
  right?: Size,
  positionStart?: Size,
  positionEnd?: Size,
  padding?: Size,
  paddingV?: Size,
  paddingH?: Size,
  paddingStart?: Size,
  paddingEnd?: Size,
  paddingLeft?: Size,
  paddingRight?: Size,
  paddingTop?: Size,
  paddingBottom?: Size,
  margin?: Size,
  marginV?: Size,
  marginH?: Size,
  marginStart?: Size,
  marginEnd?: Size,
  marginLeft?: Size,
  marginRight?: Size,
  marginTop?: Size,
  marginBottom?: Size,
  float?: 'left' | 'right' | 'start' | 'end',
  textAlign?: 'left' | 'right' | 'center' | 'start' | 'end',
  verticalAlign?: string,
  style?: Object,
};

type Context = {
  i18n: I18NContext,
};

export default function Block(
  {
    Component,
    inline,
    noWrap,
    position = 'relative',
    width,
    maxWidth,
    minWidth,
    height,
    maxHeight,
    minHeight,
    positionStart,
    positionEnd,
    top,
    left,
    bottom,
    right,
    padding,
    paddingV,
    paddingH,
    paddingStart,
    paddingEnd,
    paddingLeft,
    paddingRight,
    paddingTop,
    paddingBottom,
    margin,
    marginV,
    marginH,
    marginStart,
    marginEnd,
    marginLeft,
    marginRight,
    marginTop,
    marginBottom,
    float,
    textAlign,
    verticalAlign,
    style,
    ...props
  }: Props,
  {i18n = I18N.defaultContext}: Context,
) {
  if (Component == null) {
    Component = 'div';
  }

  if (paddingStart !== undefined) {
    if (i18n.dir === 'rtl' && paddingRight === undefined) {
      paddingRight = paddingStart;
    } else if (i18n.dir === 'ltr' && paddingLeft === undefined) {
      paddingLeft = paddingStart;
    }
  }

  if (paddingEnd !== undefined) {
    if (i18n.dir === 'rtl' && paddingLeft === undefined) {
      paddingLeft = paddingEnd;
    } else if (i18n.dir === 'ltr' && paddingRight === undefined) {
      paddingRight = paddingEnd;
    }
  }

  if (marginStart !== undefined) {
    if (i18n.dir === 'rtl' && marginRight === undefined) {
      marginRight = marginStart;
    } else if (i18n.dir === 'ltr' && marginLeft === undefined) {
      marginLeft = marginStart;
    }
  }

  if (marginEnd !== undefined) {
    if (i18n.dir === 'rtl' && marginLeft === undefined) {
      marginLeft = marginEnd;
    } else if (i18n.dir === 'ltr' && marginRight === undefined) {
      marginRight = marginEnd;
    }
  }

  if (positionStart !== undefined) {
    if (i18n.dir === 'rtl' && right === undefined) {
      right = positionStart;
    } else if (i18n.dir === 'ltr' && left === undefined) {
      left = positionStart;
    }
  }

  if (positionEnd !== undefined) {
    if (i18n.dir === 'rtl' && left === undefined) {
      left = positionEnd;
    } else if (i18n.dir === 'ltr' && right === undefined) {
      right = positionEnd;
    }
  }

  if (float === 'start') {
    float = i18n.dir === 'rtl' ? 'right' : 'left';
  } else if (float === 'end') {
    float = i18n.dir === 'rtl' ? 'left' : 'right';
  }

  if (textAlign === 'start') {
    textAlign = i18n.dir === 'rtl' ? 'right' : 'left';
  } else if (textAlign === 'end') {
    textAlign = i18n.dir === 'rtl' ? 'left' : 'right';
  }

  style = {
    paddingLeft: chooseValue(theme.padding, paddingLeft, paddingH, padding),
    paddingRight: chooseValue(theme.padding, paddingRight, paddingH, padding),
    paddingTop: chooseValue(theme.padding, paddingTop, paddingV, padding),
    paddingBottom: chooseValue(theme.padding, paddingBottom, paddingV, padding),
    marginLeft: chooseValue(theme.margin, marginLeft, marginH, margin),
    marginRight: chooseValue(theme.margin, marginRight, marginH, margin),
    marginTop: chooseValue(theme.margin, marginTop, marginV, margin),
    marginBottom: chooseValue(theme.margin, marginBottom, marginV, margin),
    display: inline ? 'inline-block' : undefined,
    whiteSpace: noWrap ? 'nowrap' : undefined,
    position,
    width,
    minWidth,
    maxWidth,
    height,
    minHeight,
    maxHeight,
    top,
    left,
    bottom,
    right,
    float,
    textAlign,
    verticalAlign,
    ...style,
  };
  return <Component {...props} style={style} />;
}

Block.contextTypes = I18N.contextTypes;
