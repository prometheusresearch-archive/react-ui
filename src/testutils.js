/**
 * @copyright 2016-present, Prometheus Research, LLC
 */

import React from 'react';
import {isEqual} from 'lodash/lang';
import indent from 'indent-string';

import {
  findWithType as _findWithType,
  findAllWithType as _findAllWithType,
  getMountedInstance,
  findAll
} from 'react-shallow-testutils';

import {
  createRenderer as _createRenderer
} from 'react-addons-test-utils';

import _inspectElement, {
  inspectReactProp as _inspectReactProp
} from 'inspect-react-element';


import assert from 'power-assert';
import {stub, spy, mock} from 'sinon';

export {stub, spy, mock, assert};

class FormDataMock {

  append = spy();
}

let originalFormData = global.FormData;

export function mockFormData() {
  global.FormData = FormDataMock;
}

export function unmockFormData() {
  global.FormData = originalFormData;
}

function inspectReactProps(node) {
  return '__originalConfig' in node.props ?
    {...node.props.__originalConfig, children: node.props.children} :
    node.props || {};
}

function inspectReactProp(propName, propValue) {
  if (propName === '__originalConfig' || propName === 'key' || propName === 'ref') {
    return null;
  }
  return _inspectReactProp(propName, propValue);
}

export function inspectElement(node) {
  return _inspectElement(node, {inspectReactProps, inspectReactProp});
}

class ReactShallowRenderer {

  constructor() {
    this._renderer = _createRenderer();
  }

  render(element, context) {
    return this._renderer.render(element, context);
  }

  get instance() {
    return getMountedInstance(this._renderer);
  }

  get element() {
    return this._renderer.getRenderOutput();
  }

  findWithType(type) {
    return findWithType(this.element, type);
  }

  findAllWithType(type) {
    return findAllWithType(this.element, type);
  }

  assertElementWithType(type) {
    this.findWithType(type);
  }

  assertNoElementWithType(type) {
    assert(this.findAllWithType(type).length === 0);
  }


  findWithTypeProps(type, props) {
    return findWithTypeProps(this.element, type, props);
  }

  findAllWithTypeProps(type, props) {
    return findAllWithTypeProps(this.element, type, props);
  }

  assertElementWithTypeProps(type, props) {
    this.findWithTypeProps(type, props);
  }

  assertNoElementWithTypeProps(type, props) {
    assert(this.findAllWithTypeProps(type, props).length === 0);
  }


  findWithElement(element) {
    return findWithElement(this.element, element);
  }

  findAllWithElement(element) {
    return findAllWithElement(this.element, element);
  }

  assertElement(element) {
    this.findWithElement(element);
  }

  assertNoElement(element) {
    assert(this.findAllWithElement(element).length === 0);
  }

}

export function createRenderer() {
  return new ReactShallowRenderer();
}

export function findWithType(root, type) {
  let elements = findAll(root, _makeMatchWithType(type));
  if (elements.length !== 1) {
    throw new Error(`Did not find exactly one partial match for element:
${indent(inspectElement(React.createElement(type)), ' ', 7)}
${indent('In the rendered element tree:', ' ', 5)}
${indent(inspectElement(root), ' ', 7)}
    `);
  }
  return elements[0];
}

export function findAllWithType(root, type) {
  return findAll(root, _makeMatchWithType(type));
}

function _makeMatchWithType(type) {
  return function _matchWithType(element) {
    return element && element.type && (
      element.type === type ||
      element.type.Component === type
    );
  };
}

export function findWithTypeProps(root, type, props) {
  let elements = findAll(root, _makeMatchWithElement(type, props));
  if (elements.length !== 1) {
    throw new Error(`Did not find exactly one (${elements.length}) partial match for element:
${indent(inspectElement(React.createElement(type, props)), ' ', 7)}
${indent('In the rendered element tree:', ' ', 5)}
${indent(inspectElement(root), ' ', 7)}
    `);
  }
  return elements[0];
}

export function findAllWithTypeProps(root, type, props) {
  return findAll(root, _makeMatchWithElement(type, props));
}

export function findWithElement(root, spec) {
  spec = _normalizeReactElementSpec(spec);
  let elements = findAll(root, _makeMatchWithElement(spec.type, spec.props));
  if (elements.length !== 1) {
    throw new Error(`Did not find exactly one (${elements.length}) partial match for element:
${indent(inspectElement(spec), ' ', 7)}
${indent('In the rendered element tree:', ' ', 5)}
${indent(inspectElement(root), ' ', 7)}
    `);
  }
  return elements[0];
}

export function findAllWithElement(root, spec) {
  spec = _normalizeReactElementSpec(spec);
  return findAll(root, _makeMatchWithElement(spec.type, spec.props));
}

function _makeMatchWithElement(type, props) {
  let matchWithType = _makeMatchWithType(type);

  let matchWithProps = function(element) {
    for (let key in props) {
      if (!props.hasOwnProperty(key)) {
        continue;
      }
      if (!element.props.hasOwnProperty(key)) {
        continue;
      }
      if (key === '__originalConfig') {
        continue;
      }
      if (!isEqual(element.props[key], props[key])) {
        return false;
      }
    }
    return true;
  };
  return function _matchWithElement(element) {
    return element && matchWithType(element) && matchWithProps(element);
  };
}

function _normalizeReactElementSpec(spec) {
  let {props: {__originalConfig = null}} = spec;
  if (__originalConfig) {
    return {...spec, props: __originalConfig};
  } else {
    return spec;
  }
}

let _createElement = React.createElement;

function createElement(type, config, ...children) {
  config = {...config, __originalConfig: config || {}};
  let element = _createElement(type, config, ...children);
  return element;
}

React.createElement = createElement;

