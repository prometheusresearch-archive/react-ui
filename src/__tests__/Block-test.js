/**
 * @copyright 2016-present, Prometheus Research, LLC
 * @flow
 */

import React from 'react';
import {createRenderer, assert} from '../testutils';
import Block from '../Block';

declare function describe(): any;
declare function it(): any;

describe('react-ui/Block', function() {

  describe('i18n props', function() {

    it('processes positionStart/positionEnd', function() {
      let renderer = createRenderer();
      renderer.render(<Block positionStart={10} positionEnd={11} />);
      assert(renderer.element.props.style.left === 10);
      assert(renderer.element.props.style.right === 11);
      renderer.render(<Block positionStart={10} positionEnd={11} />, {
        i18n: {dir: 'rtl'}
      });
      assert(renderer.element.props.style.left === 11);
      assert(renderer.element.props.style.right === 10);
    });

    it('processes paddingStart/paddingEnd', function() {
      let renderer = createRenderer();
      renderer.render(<Block paddingStart={10} paddingEnd={11} />);
      assert(renderer.element.props.style.paddingLeft === 10);
      assert(renderer.element.props.style.paddingRight === 11);
      renderer.render(<Block paddingStart={10} paddingEnd={11} />, {
        i18n: {dir: 'rtl'}
      });
      assert(renderer.element.props.style.paddingLeft === 11);
      assert(renderer.element.props.style.paddingRight === 10);
    });

    it('processes marginStart/marginEnd', function() {
      let renderer = createRenderer();
      renderer.render(<Block marginStart={10} marginEnd={11} />);
      assert(renderer.element.props.style.marginLeft === 10);
      assert(renderer.element.props.style.marginRight === 11);
      renderer.render(<Block marginStart={10} marginEnd={11} />, {
        i18n: {dir: 'rtl'}
      });
      assert(renderer.element.props.style.marginLeft === 11);
      assert(renderer.element.props.style.marginRight === 10);
    });

    it('processes textAlign="start"', function() {
      let renderer = createRenderer();
      renderer.render(<Block textAlign="start" />);
      assert(renderer.element.props.style.textAlign === 'left');
      renderer.render(<Block textAlign="start" />, {
        i18n: {dir: 'rtl'}
      });
      assert(renderer.element.props.style.textAlign === 'right');
    });

    it('processes textAlign="end"', function() {
      let renderer = createRenderer();
      renderer.render(<Block textAlign="end" />);
      assert(renderer.element.props.style.textAlign === 'right');
      renderer.render(<Block textAlign="end" />, {
        i18n: {dir: 'rtl'}
      });
      assert(renderer.element.props.style.textAlign === 'left');
    });

  });

});
