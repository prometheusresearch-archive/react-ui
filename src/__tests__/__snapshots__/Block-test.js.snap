// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n props processes float="end" 1`] = `
<div
  style={
    Object {
      "bottom": undefined,
      "display": undefined,
      "float": "right",
      "height": undefined,
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "paddingBottom": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "position": "relative",
      "right": undefined,
      "textAlign": undefined,
      "top": undefined,
      "verticalAlign": undefined,
      "whiteSpace": undefined,
      "width": undefined,
    }
  }
/>
`;

exports[`i18n props processes float="end" 2`] = `
<div
  dir="rtl"
>
  <div
    style={
      Object {
        "bottom": undefined,
        "display": undefined,
        "float": "left",
        "height": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "paddingBottom": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "position": "relative",
        "right": undefined,
        "textAlign": undefined,
        "top": undefined,
        "verticalAlign": undefined,
        "whiteSpace": undefined,
        "width": undefined,
      }
    }
  />
</div>
`;

exports[`i18n props processes float="start" 1`] = `
<div
  style={
    Object {
      "bottom": undefined,
      "display": undefined,
      "float": "left",
      "height": undefined,
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "paddingBottom": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "position": "relative",
      "right": undefined,
      "textAlign": undefined,
      "top": undefined,
      "verticalAlign": undefined,
      "whiteSpace": undefined,
      "width": undefined,
    }
  }
/>
`;

exports[`i18n props processes float="start" 2`] = `
<div
  dir="rtl"
>
  <div
    style={
      Object {
        "bottom": undefined,
        "display": undefined,
        "float": "right",
        "height": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "paddingBottom": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "position": "relative",
        "right": undefined,
        "textAlign": undefined,
        "top": undefined,
        "verticalAlign": undefined,
        "whiteSpace": undefined,
        "width": undefined,
      }
    }
  />
</div>
`;

exports[`i18n props processes marginStart/marginEnd 1`] = `
<div
  style={
    Object {
      "bottom": undefined,
      "display": undefined,
      "float": undefined,
      "height": undefined,
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": 10,
      "marginRight": 11,
      "marginTop": undefined,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "paddingBottom": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "position": "relative",
      "right": undefined,
      "textAlign": undefined,
      "top": undefined,
      "verticalAlign": undefined,
      "whiteSpace": undefined,
      "width": undefined,
    }
  }
/>
`;

exports[`i18n props processes marginStart/marginEnd 2`] = `
<div
  dir="rtl"
>
  <div
    style={
      Object {
        "bottom": undefined,
        "display": undefined,
        "float": undefined,
        "height": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": 11,
        "marginRight": 10,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "paddingBottom": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "position": "relative",
        "right": undefined,
        "textAlign": undefined,
        "top": undefined,
        "verticalAlign": undefined,
        "whiteSpace": undefined,
        "width": undefined,
      }
    }
  />
</div>
`;

exports[`i18n props processes paddingStart/paddingEnd 1`] = `
<div
  style={
    Object {
      "bottom": undefined,
      "display": undefined,
      "float": undefined,
      "height": undefined,
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "paddingBottom": undefined,
      "paddingLeft": 10,
      "paddingRight": 11,
      "paddingTop": undefined,
      "position": "relative",
      "right": undefined,
      "textAlign": undefined,
      "top": undefined,
      "verticalAlign": undefined,
      "whiteSpace": undefined,
      "width": undefined,
    }
  }
/>
`;

exports[`i18n props processes paddingStart/paddingEnd 2`] = `
<div
  dir="rtl"
>
  <div
    style={
      Object {
        "bottom": undefined,
        "display": undefined,
        "float": undefined,
        "height": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "paddingBottom": undefined,
        "paddingLeft": 11,
        "paddingRight": 10,
        "paddingTop": undefined,
        "position": "relative",
        "right": undefined,
        "textAlign": undefined,
        "top": undefined,
        "verticalAlign": undefined,
        "whiteSpace": undefined,
        "width": undefined,
      }
    }
  />
</div>
`;

exports[`i18n props processes positionStart/positionEnd 1`] = `
<div
  style={
    Object {
      "bottom": undefined,
      "display": undefined,
      "float": undefined,
      "height": undefined,
      "left": 10,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "paddingBottom": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "position": "relative",
      "right": 11,
      "textAlign": undefined,
      "top": undefined,
      "verticalAlign": undefined,
      "whiteSpace": undefined,
      "width": undefined,
    }
  }
/>
`;

exports[`i18n props processes positionStart/positionEnd 2`] = `
<div
  dir="rtl"
>
  <div
    style={
      Object {
        "bottom": undefined,
        "display": undefined,
        "float": undefined,
        "height": undefined,
        "left": 11,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "paddingBottom": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "position": "relative",
        "right": 10,
        "textAlign": undefined,
        "top": undefined,
        "verticalAlign": undefined,
        "whiteSpace": undefined,
        "width": undefined,
      }
    }
  />
</div>
`;

exports[`i18n props processes textAlign="end" 1`] = `
<div
  style={
    Object {
      "bottom": undefined,
      "display": undefined,
      "float": undefined,
      "height": undefined,
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "paddingBottom": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "position": "relative",
      "right": undefined,
      "textAlign": undefined,
      "top": undefined,
      "verticalAlign": undefined,
      "whiteSpace": undefined,
      "width": undefined,
    }
  }
  testAlign="end"
/>
`;

exports[`i18n props processes textAlign="end" 2`] = `
<div
  dir="rtl"
>
  <div
    style={
      Object {
        "bottom": undefined,
        "display": undefined,
        "float": undefined,
        "height": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "paddingBottom": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "position": "relative",
        "right": undefined,
        "textAlign": "left",
        "top": undefined,
        "verticalAlign": undefined,
        "whiteSpace": undefined,
        "width": undefined,
      }
    }
  />
</div>
`;

exports[`i18n props processes textAlign="start" 1`] = `
<div
  style={
    Object {
      "bottom": undefined,
      "display": undefined,
      "float": undefined,
      "height": undefined,
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "paddingBottom": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "position": "relative",
      "right": undefined,
      "textAlign": "left",
      "top": undefined,
      "verticalAlign": undefined,
      "whiteSpace": undefined,
      "width": undefined,
    }
  }
/>
`;

exports[`i18n props processes textAlign="start" 2`] = `
<div
  dir="rtl"
>
  <div
    style={
      Object {
        "bottom": undefined,
        "display": undefined,
        "float": undefined,
        "height": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "paddingBottom": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "position": "relative",
        "right": undefined,
        "textAlign": "right",
        "top": undefined,
        "verticalAlign": undefined,
        "whiteSpace": undefined,
        "width": undefined,
      }
    }
  />
</div>
`;
